test/e2e/loopback_test.go

```diff
@@ -0,0 +1,107 @@
+package e2e
+
+import (
+	"context"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+
+	clusterv1 "github.com/open-cluster-management/api/cluster/v1"
+	certificatesv1beta1 "k8s.io/api/certificates/v1beta1"
+
+	"github.com/open-cluster-management/registration/pkg/helpers"
+)
+
+var _ = ginkgo.Describe("Loopback registration [development]", func() {
+	ginkgo.It("Should approve CSR and SpokeCluster", func() {
+		var (
+			csrs      *certificatesv1beta1.CertificateSigningRequestList
+			err       error
+			csrClient = hubClient.CertificatesV1beta1().CertificateSigningRequests()
+		)
+
+		ginkgo.By("Waiting for the CSR to exist")
+		err = wait.Poll(1*time.Second, 45*time.Second, func() (bool, error) {
+			var err2 error
+			csrs, err2 = csrClient.List(context.TODO(), metav1.ListOptions{
+				LabelSelector: "open-cluster-management.io/cluster-name = local-development",
+			})
+			if err2 != nil {
+				return false, err2
+			}
+
+			if len(csrs.Items) >= 1 {
```

Comment:
If there are more than say five, we're hotlooping and have a bug

Reply:
I agree, do you think we should surface that here?